iFixit/Veryable
# Feature: Dockerize Veryable

# Description 

This builds off #16 aa81955 and adds a docker-compose file to now be able to run the application in a docker container. 

# Changes:
### Features
* **Dockerfile:** created dockerfile dockerignore ([3fb01f5](https://github.com/iFixit/Veryable/commit/3fb01f58270efbd8d79be7c485374cba8f8a1cac))
* **docker-compose:** created docker-compose with mysql db to start ([bb7b0b4](https://github.com/iFixit/Veryable/commit/bb7b0b4ee05a8bc7ea771f3cb175ec627dad0fcb))
* **docker-compose:** added main app service ([82feeab](https://github.com/iFixit/Veryable/commit/82feeab8ec37bdc405c8f8cc35ee281ede6403df))
* **docker-compose:** created docker start scripts ([5f4dbf2](https://github.com/iFixit/Veryable/commit/5f4dbf260604967f82a41cf5d8e0b4082afbb5ef))
* **docker-compose:** added script to run tests ([c5b70f3](https://github.com/iFixit/Veryable/commit/c5b70f3928a8745ce4084124f77cd56febb188c9))
* **docker-compose:** create prisma studio service ([83189e2](https://github.com/iFixit/Veryable/commit/83189e229a2c2e3003749f3a8b64ea29f1ac42f3))

### Misc

- changed `typescript` out directory to `dist`
- added a teardown script for `jest`

# QA

**Note:** You now need to have `Docker` installed locally to be able to spin up the containers.
You will also need the requirements specified like in https://github.com/iFixit/Veryable/pull/15#issue-1008634458

In addition, you need to update your `.env` files to match the new `.env.template`

1. Git clone this repo locally
2. Checkout this branch
3. Then run
```node
npm run start:docker
```
4. It should then start building the images
5. Once it says the containers are running validate you can run the tests on the docker container by using the following command
```node
npm run test:docker
```
6. It should have executed the tests and passed
7. Afterwards it should say `Executing teardown...` in the terminal meaning the changes from worked https://github.com/iFixit/Veryable/commit/baacff0cacb4749c82b78488cccac6628f6c8144
8. Then get the logs of the `veryable-app` container to see if it is actually running and validate `Current Pulls Today` readout shows the same number listed in the QA column of Pulldasher
9. Afterwards, check `prisma studio` is running by going to http://localhost:5555
10. Once done, try to run Veryable locally with similar steps as in https://github.com/iFixit/Veryable/pull/15#issue-1008634458

Closes #9
